build-system/tasks/check-types-build-system.js

```diff
@@ -0,0 +1,22 @@
+const { execOrThrow } = require('../common/exec');
+
+/**
+ * Performs type checking on the /build-system directory using TypeScript.
+ * Configuration is defined in /build-system/tsconfig.json.
+ *
+ * @return {Promise<void>}
+ */
+function checkBuildSystemTypes() {
+  return new Promise((resolve, reject) => {
+    try {
+      execOrThrow('npx tsc --project ./build-system/tsconfig.json', 'TypeScript build failed');
```

Comment:
Looks like `tsc` was [deprecated](https://www.npmjs.com/package/tsc) in favor of `ntypescript`, which in turn was [deprecated](https://www.npmjs.com/package/ntypescript) in favor of `byots`.

Once we get this working with the preferred tool, I think it should be explicitly added to the root `package.json` as a `devDependency`. This way, startup with `npx` will be faster because the package is already availble locally, and results will be predictable because we control (and periodically upgrade / test) the version we're using.

Reply:
I assume `tsc` was just an npm wrapper written by someone a long time ago. Looks like the best option would be to use https://www.npmjs.com/package/typescript and see if it has a native nodejs API.

Reply:
I'm just going to use the version of `tsc` within the `typescript` package